providers/celery/src/airflow/providers/celery/executors/celery_executor_utils.py

```diff
@@ -263,7 +263,7 @@ def send_task_to_executor(
     if AIRFLOW_V_3_0_PLUS:
         if TYPE_CHECKING:
             assert isinstance(args, workloads.BaseWorkload)
-        args = (args.model_dump_json(),)
+        args = (args.model_dump_json(exclude={"ti": {"executor_config"}}),)
```

Comment:
is it celery specific issue?
How can we avoid this issue for all executors? Users can roll their own executor.
I think we need to find a way to make sure the scheduler never crash, it feels like the fix is needed in the scheduler code and not only in celery?

Reply:
Yeah right now its a CE specific issue, you are right about it probably needing extension to other executors too. Would it be OK if i can handle that in a separate issue for all executors?

Reply:
> Yeah right now its a CE specific issue, you are right about it probably needing extension to other executors too.

I think the scheduler should have protection for this case on it's side regardless of what the executor does.
The scheduler should not crash. If the executor submitted a task with issue in the executor_config then the scheduler can log the exception and not schedule the task but it should not crash.

Shall I create issue for it?

Reply:
Please create an issue and assign it to me, I will take a look soon

Reply:
https://github.com/apache/airflow/issues/47377